File: 백준/Silver/11652. 카드/카드.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        long[] cards = new long[n];

        for (int i = 0; i < n; i++) {
            cards[i] = Long.parseLong(br.readLine());
        }

        Arrays.sort(cards);

        long most = cards[0];
        int maxCnt = 1;
        int curCnt = 1;

        for (int i = 1; i < n; i++) {
            if (cards[i] == cards[i - 1]) {
                curCnt++;
            } else {
                curCnt = 1;
            }

            if (curCnt > maxCnt) {
                maxCnt = curCnt;
                most = cards[i];
            }
        }

        System.out.print(most);
    }
}

File: 백준/Silver/11652. 카드/카드.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = readln().toInt()
    val arr = LongArray(n)
    for (i in 0 until n) {
        arr[i] = br.readLine().toLong()
    }
    arr.sort()
    var mxCnt = 0
    var cnt = 0
    var mxVal = -(1 shl 62) - 1L
    for (i in 0 until n) {
        if (i == 0 || arr[i - 1] == arr[i]) cnt++
        else {
            if (mxCnt < cnt) {
                mxCnt = cnt
                mxVal = arr[i - 1]
            }
            cnt = 1
        }
    }
    if (cnt > mxCnt) mxVal = arr[n - 1]
    println(mxVal)
}